persistence/sql/persister_jwk.go

```diff
@@ -84,7 +86,7 @@ func (p *Persister) AddKeySet(ctx context.Context, set string, keys *jose.JSONWe
 
 func (p *Persister) UpdateKey(ctx context.Context, set string, key *jose.JSONWebKey) error {
 	return p.transaction(ctx, func(ctx context.Context, c *pop.Connection) error {
-		if err := p.DeleteKey(ctx, set, key.KeyID); err != nil {
+		if err := p.DeleteKey(ctx, set, key.KeyID); err != nil && errorsx.Cause(err) != x.ErrNotFound {
```

Comment:
This check is new. Did something regress? Also, please use `errors.Is()`

Reply:
See my comment below (https://github.com/ory/hydra/pull/3013#discussion_r838209491)